R/designOfExperiments.R

```diff
@@ -113,11 +111,11 @@ designOfExperiments <- function(jaspResults, dataset, options, ...){
 
     if(options[["designBy"]] == "designByRuns"){
       rowIndex 	    <- which(designs[, 1] == as.numeric(options[["factorialRuns"]]))
-      resolution    <- designs[rowIndex, length(options[["factors"]])]
+      resolution    <- designs[rowIndex, as.numeric(options[["numberOfFactors"]])]
       runs          <- as.numeric(options[["factorialRuns"]])
     } else {
       resolution    <- options[["factorialResolution"]]
-      rowIndex      <- which(designs[, as.numeric(length(options[["factors"]]))] == resolution)[1]
+      rowIndex      <- which(designs[, as.numeric(options[["numberOfFactors"]])] == resolution)[1]
```

Comment:
You don't have to use `as.numeric()` here because the options is an `IntegerField`